client/network/server-url.ts

```diff
@@ -27,3 +29,17 @@ export function getServerOrigin() {
     return makeServerUrl('')
   }
 }
+
+/**
+ * Returns an URL for a path that is supposed to be in a public assets directory. In case it's not
+ * set (should be very rare), we default to our production server CDN.
+ */
+export function makePublicAssetUrl(path: string) {
+  const serverOrigin = getServerOrigin().toLowerCase()
+  const publicAssetsUrl = new JsonLocalStorageValue<string>(`${serverOrigin}:publicAssetsUrl`)
```

Comment:
The goal with this class is that you can construct it once and then use it as an accessor/writer, so you probably don't want to be constructing it on every call to this method. Make it a global variable instead (and maybe put it somewhere common that both the thing loading the config + this can use).